core/capabilities/ccip/ccipsolana/commitcodec_test.go

```diff
@@ -0,0 +1,133 @@
+	pubkey, err := solanago.NewRandomPrivateKey()
+	if err != nil {
+		panic(err)
+	}
+
+	return cciptypes.CommitPluginReport{
+		MerkleRoots: []cciptypes.MerkleRootChain{
+			{
+				OnRampAddress: cciptypes.UnknownAddress(pubkey.PublicKey().String()),
+				ChainSel:      cciptypes.ChainSelector(rand.Uint64()),
+				SeqNumsRange: cciptypes.NewSeqNumRange(
+					cciptypes.SeqNum(rand.Uint64()),
+					cciptypes.SeqNum(rand.Uint64()),
+				),
+				MerkleRoot: utils.RandomBytes32(),
+			},
+		},
+		PriceUpdates: cciptypes.PriceUpdates{
+			TokenPriceUpdates: []cciptypes.TokenPrice{
+				{
+					TokenID: "C8WSPj3yyus1YN3yNB6YA5zStYtbjQWtpmKadmvyUXq8",
+					Price:   cciptypes.NewBigInt(big.NewInt(rand.Int63())),
+				},
+			},
+			GasPriceUpdates: []cciptypes.GasPriceChain{
+				{GasPrice: cciptypes.NewBigInt(big.NewInt(rand.Int63())), ChainSel: cciptypes.ChainSelector(rand.Uint64())},
+				{GasPrice: cciptypes.NewBigInt(big.NewInt(rand.Int63())), ChainSel: cciptypes.ChainSelector(rand.Uint64())},
+				{GasPrice: cciptypes.NewBigInt(big.NewInt(rand.Int63())), ChainSel: cciptypes.ChainSelector(rand.Uint64())},
+			},
+		},
+	}
+}
+
+func TestCommitPluginCodecV1(t *testing.T) {
+	testCases := []struct {
+		name   string
+		report func(report cciptypes.CommitPluginReport) cciptypes.CommitPluginReport
+		expErr bool
+	}{
+		{
+			name: "base report",
+			report: func(report cciptypes.CommitPluginReport) cciptypes.CommitPluginReport {
+				return report
+			},
+		},
+		{
+			name: "empty token address",
+			report: func(report cciptypes.CommitPluginReport) cciptypes.CommitPluginReport {
+				report.PriceUpdates.TokenPriceUpdates[0].TokenID = ""
+				return report
+			},
+			expErr: true,
+		},
+		{
+			name: "empty merkle root",
+			report: func(report cciptypes.CommitPluginReport) cciptypes.CommitPluginReport {
+				report.MerkleRoots[0].MerkleRoot = cciptypes.Bytes32{}
+				return report
+			},
+		},
+		{
+			name: "zero token price",
+			report: func(report cciptypes.CommitPluginReport) cciptypes.CommitPluginReport {
+				report.PriceUpdates.TokenPriceUpdates[0].Price = cciptypes.NewBigInt(big.NewInt(0))
+				return report
+			},
+		},
+		{
+			name: "zero gas price",
+			report: func(report cciptypes.CommitPluginReport) cciptypes.CommitPluginReport {
+				report.PriceUpdates.GasPriceUpdates[0].GasPrice = cciptypes.NewBigInt(big.NewInt(0))
+				return report
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			report := tc.report(randomCommitReport())
+			commitCodec := NewCommitPluginCodecV1()
+			ctx := testutils.Context(t)
+			encodedReport, err := commitCodec.Encode(ctx, report)
+			if tc.expErr {
+				assert.Error(t, err)
+				return
+			}
+			require.NoError(t, err)
+			decodedReport, err := commitCodec.Decode(ctx, encodedReport)
+			require.NoError(t, err)
+			require.Equal(t, report, decodedReport)
+		})
+	}
+}
+
+func BenchmarkCommitPluginCodecV1_Encode(b *testing.B) {
```

Comment:
what are these Benchmark unused functions?

Could we have 1 test that encodes a report, decodes it back, and compares that original matches exactly the one after encoding and decoding?

Reply:
It's for benchmarking purpose similar to what evm codec test had, it measures ns/op speed etc.

I added another benchmark for decode encoded report